#!/usr/bin/env python
PACKAGE = "roller_eye"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#  the param of image from camera
# name, type, level,description, default, min, max
gen.add("image_width",  int_t,    0, "width", 1280,  100, 4000)
gen.add("image_height", int_t,    0, "height", 720,  100, 4000)
gen.add("image_fps_num",    int_t,    0, "1/fps numerator", 1,  1, 10)
gen.add("image_fps_den",    int_t,    0, "1/fps denominator", 30,  1, 120)
gen.add("image_fmt",    int_t,    0, "frame fmt,0:yuv420,1:yuyv", 0,  0, 10)  #deprecated
gen.add("image_night_mode",    int_t,    0, "night mode,0:off,1:on,2:auto", 2,  0, 2)
gen.add("wideDynamic",    int_t,    0, "wide dynamic,0:auto,1:hand", 0,  0,1)
gen.add("wideDynamicRange",    int_t,    0, "wide dynamic range", 3,  0, 5)
gen.add("h264Quality",    int_t,    0, "h264 Compression multiple", 1,  0, 2)
gen.add("nightVisionSensitivity",    int_t,    0, "night vision sensitivity", 0,  0, 1)
gen.add("cameraLight",    int_t,    0, "camera brightness", 0,  0, 100)

exit(gen.generate(PACKAGE, "ParamNode", "Image"))